include/deal.II/sundials/sunlinsol_newempty.h

```diff
@@ -0,0 +1,146 @@
+//-----------------------------------------------------------
+//
+//    Copyright (C) 2020 by the deal.II authors
+//
+//    This file is part of the deal.II library.
+//
+//    The deal.II library is free software; you can use it, redistribute
+//    it, and/or modify it under the terms of the GNU Lesser General
+//    Public License as published by the Free Software Foundation; either
+//    version 2.1 of the License, or (at your option) any later version.
+//    The full text of the license can be found in the file LICENSE.md at
+//    the top level directory of deal.II.
+//
+//-----------------------------------------------------------
+
+/**
+ * This file contains a verbatim copy of code snippets distributed within the
+ * SUNDIALS package, see the license below.
+ */
```

Comment:
As already mentioned I had to copy two functions from the newest SUNDIALS version to use 4.0.0. I put them in this file since I have no idea how to handle such a case especially with regards to licenses.